app/_layouts/shared-layout.tsx

```diff
@@ -79,6 +81,11 @@ export default async function SharedLayout({
   const dictionary = await getDictionary(lang);
   const pageMap = await getPageMap(`/${lang}`);
 
+  // Get the current pathname to determine initial sidebar state
+  const headersList = await headers();
+  const pathname = headersList.get("x-pathname") || "";
+  const isHomePage = pathname.endsWith("/home");
```

Comment:
We also need check on ssr if we are on the homepage or not to start with the sidebar collapsed or not, otherwise you see a jank animation post-pageload.